build-tools/packages/build-cli/src/commands/check/prApproval.ts

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Flags } from "@oclif/core";
+
+import { githubActionsFlag, githubTokenFlag } from "../../flags.js";
+
+import {
+	type GitHubProps,
+	isPrApprovedByTeam,
+	isPrApprovedByUsers,
+	// We are slowly moving away from barrel files and these APIs are only used here.
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../library/githubRest.js";
+import { BaseCommand } from "../../library/index.js";
+
+/**
+ * This command class is used to merge two branches based on the batch size provided.
+ * It looks for the last common commit between two branches and computes the remaining commits to be merged.
+ * Later, it creates a pull request based on the batch size passed.
+ */
+export default class CheckPrApprovalCommand extends BaseCommand<
```

Comment:
If we were to want to enforce multiple review requirements for the same conditions, what would the flow look like for using this (e.g., if multiple teams were required)? Looks like currently you would have to call this separately for each reviewer group? Wondering if this is a case we should consider here.

Reply:
Yes, it only checks one group so you'd have to call it multiple times. Seems easy enough to update the command to take multiple teams, or multiple groups, or whatever, should requirements
arise. That said, doing them separately can make for a better report UX for the CI run.